### DISCUSSION TITLE
.SECTION seems to alter output in inexplicable ways?

### DISCUSSION THREAD

**Discussion by Pokeman2003** [vhelin/wla-dx#574] [Q&A] [answered]
Okay, so I'm very tired so I'll keep it brief. Decided to try out .SECTION to label the things. Everything worked, though there seemed to be slight disturbances with the operation. Then I decided to add .SECTION "Initalization", and all hell broke loose. 
It assembles and links perfectly fine according to the tools, but the output does not work with Gearsystem 3.4, creating this odd flickering effect.
Included below are my source codes(functional and non-functional respectively), and the compile script I used(it's a .bat script.)
[Functional Build.zip](https://github.com/vhelin/wla-dx/files/11414782/Functional.Build.zip)
[Non Functional Build.zip](https://github.com/vhelin/wla-dx/files/11414783/Non.Functional.Build.zip)
[Compile.txt](https://github.com/vhelin/wla-dx/files/11414789/Compile.txt)
Why is this happening, how can I fix this, and since I KNOW what I'm doing isn't the correct way to do this, how am I supposed to be using .SECTION?

**Comment by vhelin** [comment]
Without looking at the code (no time to investigate what a Gearsystem 3.4 is and why your code doesn't work with it at the moment, perhaps later next week?) perhaps your issue is related to the types (FREE, FORCE, SEMIFREE...) of the .SECTION as they affect the placing of the .SECTION into the output. 

Does your code work without .SECTIONs? If you add .SECTIONs but make them FORCE then the result should be identical to if you didn't use .SECTIONs at all...

Any Gearsystem users out here?

**Comment by vhelin** [comment, marked as answer]
I see you are using .SECTIONs without specifying their types so they are FREE by default which makes the linker to sort them by their size and place them into suitable spots in the output. If you want to keep e.g. this .SECTION at ORG 0 then you need to add keyword FORCE to the .SECTION definition...

![kuva](https://user-images.githubusercontent.com/6417746/236678542-b78ae92b-7c15-4e97-a332-37b8f34723d8.png)

**Reply by Pokeman2003** [reply]
Oh wow, thanks! I didn't realize this would happen. I guess that's the problem with trying new things out at 2 AM.
